### PR TITLE
Vendors

### PR DESCRIPTION
**Pull request by RomaKushnir**

### CODE REVIEW

**Comment by RomaKushnir** [comment]
Created general AddNewItemButton component
Created general ItemActionButton component with types to define different styles
Added files with mock data
Removed Header component props and changed routes path due to constants in file router.js
Created Vendors page with it's components

File: src/components/AddNewItemButton/AddNewItemButton.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import styles from './AddNewItemButton.module.scss';
+
+function AddNewItemButton({
+  btnTitle,
+  onAddNewItem
+}) {
+  return (
+    <button
+      type="button"
+      className={styles['add-btn']}
```
Comment (artemtool) [inline src/components/AddNewItemButton/AddNewItemButton.js:11]: Please, use camelCase for classNames.

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -3,8 +3,10 @@ import styles from './Footer.module.scss';
 function Footer() {
   return (
     <footer className={styles['footer-wrapper']}>
```
Comment (artemtool) [inline src/components/Footer/Footer.js:5]: Please, use camelCase for classNames.

File: src/components/Footer/Footer.js [hunk 3]
```diff
@@ -3,8 +3,10 @@ import styles from './Footer.module.scss';
...
-      <p className={styles.slogan}>Save your money with Exadel</p>
-      <p className={styles.copyright}>Copyright &copy; 2021 All rights reserved.</p>
+      <div className={styles['footer-content']}>
```
Comment (artemtool) [inline src/components/Footer/Footer.js:6]: Please, use camelCase for classNames.

File: src/components/ItemActionButton/ItemActionButton.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  title,
+  type = 'normal',
+  isDisabled = false,
+  className = '',
+  onActionClick
+}) {
+  return (
+    <button
+      type="button"
+      className={`${styles['action-btn']} ${styles[type]} ${className}`}
```
Comment (artemtool) [inline src/components/ItemActionButton/ItemActionButton.js:13]: Please, use camelCase for classNames.

File: src/pages/Vendors/Vendors.js [hunk 5]
```diff
@@ -15,22 +26,60 @@ function Vendors() {
...
-      <Modal
-        isOpen = {isOpen}
-        onClose = {() => setIsOpen(false)}
-      >
-      <AddVendor onSave = {() => setIsOpen(false)}/>
-      </Modal>
+    <div className={styles.container}>
+      <div>
+        <Header />
+        <main className={styles['content-wrapper']}>
```
Comment (artemtool) [inline src/pages/Vendors/Vendors.js:50]: Please, use camelCase for classNames.

File: src/pages/Vendors/Vendors.js [hunk 6]
```diff
@@ -15,22 +26,60 @@ function Vendors() {
...
+          <FiltersContainer
+            onApplyButtonClick={onApplyButtonClick}
+            countriesList={countriesList}
+            citiesList={citiesList}
+            categoriesList={[]}
+            vendorsList={vendorsList}
+          />
+          <div className={styles['vendors-actions-block']}>
```
Comment (artemtool) [inline src/pages/Vendors/Vendors.js:58]: Please, use camelCase for classNames.

File: src/pages/Vendors/Vendors.js [hunk 7]
```diff
@@ -15,22 +26,60 @@ function Vendors() {
...
+              options={sortList}
+              onChange={onSortFilter}
+              isClearable={false}
+              />
+          </div>
+          <Modal isOpen={isOpen} onClose={() => setIsOpen(false)}>
+            <AddVendorModal onSave={() => setIsOpen(false)} />
+          </Modal>
+          <VendorsList vendors={VendorsListPage} />
+          <div className={styles['show-more-btn-wrap']}>
```
Comment (artemtool) [inline src/pages/Vendors/Vendors.js:74]: Please, use camelCase for classNames.

File: src/pages/Vendors/components/VendorListItem/VendorListItem.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  const onEdit = () => {
+    // do code to edit item
+  };
+
+  const onDelete = () => {
+    // do code to delete item
+  };
+
+  return (
+    <div className={styles['item-wrapper']}>
```
Comment (artemtool) [inline src/pages/Vendors/components/VendorListItem/VendorListItem.js:14]: Please, use camelCase for classNames.

File: src/pages/Vendors/components/VendorListItem/VendorListItem.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      <figure className={styles['img-wrapper']}>
```
Comment (artemtool) [inline src/pages/Vendors/components/VendorListItem/VendorListItem.js:15]: Please, use camelCase for classNames.

File: src/pages/Vendors/components/VendorListItem/VendorListItem.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        <img src={vendor.imageUrl} alt={vendor.title} />
+      </figure>
+      <div className={styles['text-wrapper']}>
+        <p className={styles['vendor-title']}>{vendor.title}</p>
+        <p className={styles['vendor-location']}>{vendor.location}</p>
+        <p className={styles['vendor-description']}>{vendor.description}</p>
```
Comment (artemtool) [inline src/pages/Vendors/components/VendorListItem/VendorListItem.js:21]: Please, use camelCase for classNames.

File: src/pages/Vendors/components/VendorsList/VendorsList.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import VendorListItem from '../VendorListItem';
+
+function VendorsList({ vendors }) {
+  return (
+    <div>
+      {vendors.length
+        ? vendors.map(
+          (vendor) => <VendorListItem vendor={vendor} key={vendor.id} />
+        )
+        : <p style={ { textAlign: 'center', marginTop: '10vh' } }>There are no vendors yet!</p>
```
Comment (artemtool) [inline src/pages/Vendors/components/VendorsList/VendorsList.js:10]: Put styles into stylesheet and use className to style "p" tag.

**Review by artemtool — APPROVED** [review]

**Pull request merged**